From b39ed158af1bcbd8a11e3bd4224ce01011662e57 Mon Sep 17 00:00:00 2001
Message-ID: <b39ed158af1bcbd8a11e3bd4224ce01011662e57.1747779547.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1747779547.git.sharlatanus@gmail.com>
References: <cover.1747779547.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 12 May 2025 18:54:12 +0100
Subject: [PATCH 07/31] gnu: Add python-globus-sdk.

* gnu/packages/python-web.scm (python-globus-sdk): New variable.

Change-Id: Idb8ce6d9bc96ba30615e6de9a3b5612489ab47a8
---
 gnu/packages/python-web.scm | 42 +++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/gnu/packages/python-web.scm b/gnu/packages/python-web.scm
index 56f697bbd2..a100a6cd60 100644
--- a/gnu/packages/python-web.scm
+++ b/gnu/packages/python-web.scm
@@ -552,6 +552,48 @@ (define-public python-eventlet
 from the Python interpreter, or as a small part of a larger application.")
     (license license:expat)))
 
+(define-public python-globus-sdk
+  (package
+    (name "python-globus-sdk")
+    (version "3.56.0")
+    (source
+     (origin
+       (method git-fetch)               ;no tests in PyPI archive
+       (uri (git-reference
+             (url "https://github.com/globus/globus-sdk-python")
+             (commit version)))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "11nljda2ir4gna4xa5vkj5nzxnjwadkh97qplkk9nrj44szphnzw"))))
+    (build-system pyproject-build-system)
+    (arguments
+     (list
+      #:test-flags
+      #~(list "--numprocesses" (number->string (min 8 (parallel-job-count))))
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-before 'check 'pre-check
+            (lambda _ (setenv "HOME" "/tmp"))))))
+    (native-inputs
+     (list python-flaky
+           python-pytest
+           python-pytest-randomly
+           python-pytest-xdist
+           python-responses
+           python-setuptools
+           python-wheel))
+    (propagated-inputs
+     (list python-cryptography
+           python-importlib-resources
+           python-pyjwt
+           python-requests))
+    (home-page "https://github.com/globus/globus-sdk-python")
+    (synopsis "Globus SDK for Python")
+    (description
+     "This package provides a SDK for convenient Pythonic interface to
+@url{https://www.globus.org/, Globus} APIs.")
+    (license license:asl2.0)))
+
 (define-public python-hookdns
   (package
     (name "python-hookdns")
-- 
2.49.0

